src/test/java/dog/pawbook/logic/parser/DeleteDogCommandParserTest.java

```diff
@@ -0,0 +1,33 @@
+package dog.pawbook.logic.parser;
+
+import static dog.pawbook.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static dog.pawbook.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static dog.pawbook.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static dog.pawbook.testutil.TypicalIndexes.INDEX_FIRST_DOG;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.logic.commands.DeleteDogCommand;
+
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the DeleteCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the DeleteCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+public class DeleteDogCommandParserTest {
+
+    private DeleteDogCommandParser parser = new DeleteDogCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsDeleteCommand() {
+        assertParseSuccess(parser, "1", new DeleteDogCommand(INDEX_FIRST_DOG));
```

Comment:
Tbh I'm quite keen to remove the use of INDEX_*_*, because we have switched to referencing by ID and not by the visible index number. For this method it is fine of course, but there is a need to probably remove Index and create an ID class in its place, but that's probably another issue to be created.

Reply:
Not to mention the convoluted construction and usage of the Index class doesn't seem to help.

Reply:
yea i agree that the INDEX_ references are not that necessary